aws/resource_aws_msk_cluster.go

```diff
@@ -587,6 +586,41 @@ func resourceAwsMskClusterUpdate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if d.HasChange("kafka_version") {
+		var input *kafka.UpdateClusterKafkaVersionInput
+
+		if d.HasChange("configuration_info") {
+			input = &kafka.UpdateClusterKafkaVersionInput{
+				ClusterArn:         aws.String(d.Id()),
+				ConfigurationInfo:  expandMskClusterConfigurationInfo(d.Get("configuration_info").([]interface{})),
+				CurrentVersion:     aws.String(d.Get("current_version").(string)),
+				TargetKafkaVersion: aws.String(d.Get("kafka_version").(string)),
+			}
+		} else {
+			input = &kafka.UpdateClusterKafkaVersionInput{
+				ClusterArn:         aws.String(d.Id()),
+				CurrentVersion:     aws.String(d.Get("current_version").(string)),
+				TargetKafkaVersion: aws.String(d.Get("kafka_version").(string)),
+			}
+		}
```

Comment:
Nit: can consolidate the shared input values to:
```suggestion
		input := &kafka.UpdateClusterKafkaVersionInput{
			ClusterArn:         aws.String(d.Id()),
			CurrentVersion:     aws.String(d.Get("current_version").(string)),
			TargetKafkaVersion: aws.String(d.Get("kafka_version").(string)),
		}

		if d.HasChange("configuration_info") {
			input.ConfigurationInfo = expandMskClusterConfigurationInfo(d.Get("configuration_info").([]interface{}))
		}
```